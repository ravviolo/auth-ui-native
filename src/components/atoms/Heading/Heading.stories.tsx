import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { Heading } from './Heading';

storiesOf('Heading', module)
  .addDecorator((getStory) => <StoryScreen center>{getStory()}</StoryScreen>)
  .add('Default', () => (
    <Heading
      testID={text('testID', 'heading-test-id')}
      title={text('title', 'This is a heading')}
    />
  ));
