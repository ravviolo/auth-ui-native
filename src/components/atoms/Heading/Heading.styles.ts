import { StyleSheet } from 'react-native';
import { colors } from 'theme';

export const styles = StyleSheet.create({
  heading: {
    color: colors.text,
    fontSize: 18,
    fontWeight: 'bold',
    textTransform: 'uppercase',
  },
});
