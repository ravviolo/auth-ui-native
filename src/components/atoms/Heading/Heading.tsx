import { StyleSheet, Text } from 'react-native';
import { colors } from 'theme';

interface Props {
  title: string;
  testID: string;
}

export const Heading = ({ title, testID }: Props) => {
  return (
    <Text style={styles.heading} testID={testID}>
      {title}
    </Text>
  );
};

const styles = StyleSheet.create({
  heading: {
    color: colors.text,
    fontSize: 18,
    fontWeight: 'bold',
    textTransform: 'uppercase',
  },
});
