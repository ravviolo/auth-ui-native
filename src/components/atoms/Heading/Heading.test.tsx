import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Heading } from './Heading';

import type { Props } from './Heading';

describe('Heading', () => {
  const props: Props = {
    testID: 'heading-test-id',
    title: 'Title',
  };

  it('should display text received via props', () => {
    render(<Heading {...props} />);

    const heading = screen.getByTestId('heading-test-id');

    expect(heading).toHaveTextContent('Title');
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<Heading {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
