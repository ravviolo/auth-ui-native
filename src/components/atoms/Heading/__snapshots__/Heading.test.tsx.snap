// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should match snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#3D3D3D",
      "fontSize": 18,
      "fontWeight": "bold",
      "textTransform": "uppercase",
    }
  }
  testID="heading-test-id"
>
  Title
</Text>
`;
